Tentacle/Client.swift

```diff
@@ -390,6 +393,54 @@ public final class Client {
                     .concat(response.links["next"] == nil ? SignalProducer.empty : self.fetchMany(endpoint, page: nextPage, pageSize: pageSize))
             }
     }
+
+    internal func send<Request: RequestType>(_ request: Request, to endpoint: Endpoint, method: String) -> SignalProducer<(Response, Request.Response), Error> where Request.Response == Request.Response.DecodedType {
+        let url = URL(server, endpoint)
+
+        var urlRequest = URLRequest.create(url, credentials)
+        urlRequest.httpMethod = method
+
+        let object = request.encode().JSONObject()
+        if let payload = try? JSONSerialization.data(withJSONObject: object, options: []) {
+            urlRequest.httpBody = payload
+        }
+
+        return urlSession
+            .reactive
+            .data(with: urlRequest)
+            .mapError(Error.networkError)
+            .flatMap(.concat) { data, response -> SignalProducer<(Response, Request.Response), Error> in
+                let response = response as! HTTPURLResponse
+                let headers = response.allHeaderFields as! [String:String]
+
+                return SignalProducer.attempt {
+                    return JSONSerialization
+                        .deserializeJSON(data)
+                        .mapError { Error.jsonDeserializationError($0.error) }
+                }
+                .attemptMap { JSON -> Result<Any, Client.Error> in
+                    if response.statusCode == 404 {
+                        return .failure(.doesNotExist)
+                    }
+                    if response.statusCode >= 400 && response.statusCode < 600 {
+                        return decode(JSON)
+                            .mapError(Error.jsonDecodingError)
+                            .flatMap { error in
+                                .failure(Error.apiError(response.statusCode, Response(headerFields: headers), error))
+                        }
+                    }
+                    return .success(JSON)
+
+                }
+                .attemptMap { j in
+                    return decode(j)
+                        .mapError(Error.jsonDecodingError)
+                }
+                .map {
+                    return (Response(headerFields: headers), $0)
+                }
+            }
```

Comment:
These are a lot of code here that's shared with `fetch`. We should somehow combine or share code between them.